app/controllers/category_experts_controller.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class CategoryExpertsController < ApplicationController
+
+  def endorse
+    raise Discourse::NotFound unless current_user
+    user = fetch_user_from_params
+
+    category_ids = params[:categoryIds]&.reject(&:blank?)
+
+    raise Discourse::InvalidParameters if !category_ids || category_ids.blank?
```

Comment:
```suggestion
    raise Discourse::InvalidParameters if category_ids.blank?
```
The first check should not be necessary.